python/cudf/cudf/pylibcudf_tests/test_column_factories.py

```diff
@@ -0,0 +1,182 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import pyarrow as pa
+import pytest
+from utils import assert_column_eq
+
+from cudf._lib import pylibcudf as plc
+
+
+@pytest.fixture(
+    scope="module",
+    params=[
+        pa.uint8(),
+        pa.uint16(),
+        pa.uint32(),
+        pa.uint64(),
+        pa.int8(),
+        pa.int16(),
+        pa.int32(),
+        pa.int64(),
+        pa.float32(),
+        pa.float64(),
+    ],
+    ids=[
+        "uint8",
+        "uint16",
+        "uint32",
+        "uint64",
+        "int8",
+        "int16",
+        "int32",
+        "int64",
+        "float32",
+        "float64",
+    ],
+)
+def numeric_pa_type(request):
+    return request.param
+
+
+@pytest.fixture(
+    scope="module",
+    params=[
+        pa.timestamp("s"),
+        pa.timestamp("ms"),
+        pa.timestamp("us"),
+        pa.timestamp("ns"),
+    ],
+    ids=["s", "ms", "us", "ns"],
+)
+def timestamp_pa_type(request):
+    return request.param
+
+
+@pytest.fixture(
+    scope="module",
+    params=[
+        pa.duration("s"),
+        pa.duration("ms"),
+        pa.duration("us"),
+        pa.duration("ns"),
+    ],
+    ids=["s", "ms", "us", "ns"],
+)
+def duration_pa_type(request):
+    return request.param
+
+
+@pytest.fixture(
+    scope="module",
+    params=[
+        plc.column_factories.MaskState.UNALLOCATED,
+        plc.column_factories.MaskState.ALL_VALID,
+        plc.column_factories.MaskState.ALL_NULL,
+        plc.column_factories.MaskState.UNINITIALIZED,
+    ],
+    ids=["unallocated", "all_valid", "all_null", "uninitialized"],
+)
+def mask_state(request):
+    return request.param
+
+
+def test_make_empty_column_dtype(pa_type):
+    pa_col = pa.array([], type=pa_type)
+
+    # TODO: DataType.from_arrow()?
```

Comment:
Should `plc.interop.from_arrow` handle pyarrow type objects as well? For other objects it seems to rely on `cpp_from_arrow` and does some inspection of the result to get the type, so we might need to go a different route. It might not need to be much more complicated than a simple dict lookup.

Reply:
Yup I think this is the right idea. I wrote up this proposal to the previous TODO message, moving it here.

I think that we should we go ahead and implement a `plc.interop.from_arrow(arrow_type)` function. That function can mostly be just a lookup, with a few exceptions. I'm picturing something like

```
default_mapping = {
....
}
@from_arrow.register(pa.DataType)
from_arrow(pyarrow_type):
    try:
        return default_mapping.get(pyarrow_type)
    except KeyError:
        # For special cases like decimal where we need to do additional processing and not just a dict lookup
```

Reply:
Added 👍